src/lib/utils/index.ts

```diff
@@ -292,3 +293,15 @@ export function simpleDebounce<T extends (...args: any[]) => Promise<any>>(fn: T
 export type Index<T> = {
   [key: string]: T
 }
+
+/**
+ * An ESM-aware reading of the main entrypoint to a package.
+ */
+export function getPackageJsonMain(packageJson: PackageJson & { main: string }): string {
+  // todo when building for a bundler, we want to read from the esm paths. Otherwise the cjs paths.
+  //  - this condition takes a stab at the problem but is basically a stub.
+  //  - this todo only needs to be completed once we are actually trying to do esm tree-shaking (meaning, we've moved beyond node-file-trace)
```

Comment:
Yeah... And in the real future it'll also need to deal with .mjs which is through the `package.export` field.